src/main/java/org/opentripplanner/graph_builder/module/osm/ParkingProcessor.java

```diff
@@ -275,6 +279,45 @@ private VehicleParking buildParkAndRideAreasForGroup(AreaGroup group, boolean is
     return vehicleParking;
   }
 
+  /**
+   * Creates an artificial entrance to a parking facility's centroid.
+   * <p>
+   * This is useful if the facility is not linked to the street network in OSM. Without this method
+   * it would not be usable by the routing algorithm as it's unreachable.
+   */
+  private List<VehicleParking.VehicleParkingEntranceCreator> createArtificialEntrances(
+    AreaGroup group,
+    I18NString vehicleParkingName,
+    OSMWithTags entity,
+    boolean isCarPark
+  ) {
+    var geoms = group.areas.stream().map(a -> a.jtsMultiPolygon).toArray(Geometry[]::new);
```

Comment:
We could store `u` as `geometry` or something similar for the AreaGroup [here](https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/main/java/org/opentripplanner/graph_builder/module/osm/AreaGroup.java#L68), and not need to recompute it again.

Reply:
Don't `AreaGroups` end up in the graph so this would increase memory, wouldn't it?

Reply:
No, `AreaGroups` is not part of the graph

Reply:
They are used only for the input parameter for `ParkingProcessor` and in `buildWalkableAreas`.

Reply:
I've implemented it. Did you want to store the multipolygon or the union of it?